integration-tests/src/main/java/org/apache/druid/cli/MainForQueryRetryTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.cli;
+
+import com.google.inject.Injector;
+import io.airlift.airline.Cli;
+import io.airlift.airline.Help;
+import io.airlift.airline.ParseException;
+import io.netty.util.SuppressForbidden;
+import org.apache.druid.guice.ExtensionsConfig;
+import org.apache.druid.guice.GuiceInjectors;
+import org.apache.druid.initialization.Initialization;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+
+/**
+ * Main class for query retry testing.
+ *
+ * @see CliHistoricalForQueryRetryTest
+ */
+public class MainForQueryRetryTest
```

Comment:
Thanks for pointing it out. I didn't know that is a thing. However, looking at that, it seems like a useful way to develop a custom service based on Druid's extension system. This will work for now since the integration tests load all extensions with the same class loader, but won't work once we fix them to use Druid's extension loading system. I would keep the current implementation for now.

Reply:
Hmm, first of all, I didn't notice that our integration tests already use Druid's extension system. This is a pretty good improvement! 🙂 Looking at this more, the suggested way seems working since it's the main thread which loads and executes this command. I have added an integration-tests extension and made it to be loaded for `query-retry` testing group.